### PR TITLE
[draft] feat(form-core): default fieldNames and articles

### PR DESCRIPTION
**Pull request by tlouisse**
Proposal for default fieldNames in validation (instead of label name).

Additionally, we could consider:

- adding default fieldNames `'a date'|'an email'|'an iban'|'an amount'` for lion-input-(date|email|iban|amount)

### CODE REVIEW

File: packages/form-core/test-suites/ValidateMixinFeedbackPart.suite.js [hunk 1]
```diff
@@ -517,5 +577,32 @@ export function runValidateMixinFeedbackPart() {
...
+        <${tag}
+          .validators=${[new Required()]}
+        >${lightDom}</${tag}>
+      `);
+        await el.feedbackComplete;
+
+        expect(el._feedbackNode.feedbackData[0].message).to.equal('Fill in something');
+      });
+
+      it('can set global validator messages', async () => {
```
Comment (tlouisse) [inline packages/form-core/test-suites/ValidateMixinFeedbackPart.suite.js:594]: `can set global fieldNames and articles`

File: packages/form-core/test-suites/ValidateMixinFeedbackPart.suite.js [hunk 2]
```diff
@@ -517,5 +577,32 @@ export function runValidateMixinFeedbackPart() {
...
+        Required.getMessage = ({ article, fieldName }) => `Please enter ${article} ${fieldName}`;
+        ValidateMixin.fieldName = 'value';
```
Comment (tlouisse) [inline packages/form-core/test-suites/ValidateMixinFeedbackPart.suite.js:596]: ValidateMixin is a function, not a constructor. This should be considered pseudo code

File: packages/validate-messages/translations/en.js [hunk 3]
```diff
@@ -49,4 +49,9 @@ export default {
     Changed: 'Changed!',
     OkCorrect: 'Ok, correct.',
   },
+  defaultSubject: {
```
Comment (tlouisse) [inline packages/validate-messages/translations/en.js:52]: Make this:

```
defaultFieldName: {
  field: 'value',
  article: 'a', // {'a'|'an'} 
  gender: 'm' // {'m'|'f'|'n'} => optional, only important for French etc.
},

```

File: packages/form-core/src/validate/ValidateMixin.js [hunk 4]
```diff
@@ -514,15 +539,21 @@ export const ValidateMixin = dedupeMixin(
...
       async __getFeedbackMessages(validators) {
         let fieldName = await this.fieldName;
+        let article = await this.article;
+
         return Promise.all(
           validators.map(async validator => {
             if (validator.config.fieldName) {
               fieldName = await validator.config.fieldName;
             }
+            if (validator.config.article) {
```
Comment (tlouisse) [inline packages/form-core/src/validate/ValidateMixin.js:549]: fieldName can be both a string or object, see `defaultFieldName` below...

File: packages/validate-messages/translations/en.js (hunk 3 above)
Comment (tlouisse) [reply packages/validate-messages/translations/en.js:52]: => call it `langParams`?

File: packages/form-core/test-suites/ValidateMixinFeedbackPart.suite.js [hunk 5]
```diff
@@ -517,5 +517,31 @@ export function runValidateMixinFeedbackPart() {
 
       expect(el._feedbackNode.feedbackData).to.deep.equal([]);
     });
+
+    describe('Global feedback messages', () => {
+      it('can set global validator messages', async () => {
```
Comment (tlouisse) [inline packages/form-core/test-suites/ValidateMixinFeedbackPart.suite.js:522]: Not directly related to this new feature, just a missing test ;)

File: packages/form-core/test-suites/ValidateMixinFeedbackPart.suite.js [hunk 6]
```diff
@@ -459,36 +459,36 @@ export function runValidateMixinFeedbackPart() {
...
         });
       });
-    });
 
-    it('".getMessage()" gets .fieldName defined on Validator config', async () => {
-      const constructorValidator = new MinLength(4, {
-        fieldName: new Promise(resolve => resolve('myFieldViaCfg')),
-      });
-      const spy = sinon.spy(constructorValidator.constructor, 'getMessage');
+      it('".getMessage()" gets .fieldName defined on Validator config', async () => {
```
Comment (tlouisse) [inline packages/form-core/test-suites/ValidateMixinFeedbackPart.suite.js:463]: nothing going on here, only a test moved into a describe ;)

**Comment by jorenbroekema** [comment]
Closing this for now. Perhaps open a separate PR for default field names if we still want to have that one.

**Pull request closed without merging**